apps/schematic/api/schematic_api/controllers/schema_controller_impl.py

```diff
@@ -340,25 +346,26 @@ def get_node_dependencies(
           the schema, otherwise random
 
     Returns:
-        list[str]: A list of node labels or display names
+        list[Node]: A list of nodes
 
     """
     schema_generator = SchemaGenerator(path_to_json_ld=schema_url)
-    return schema_generator.get_node_dependencies(
+    nodes = schema_generator.get_node_dependencies(
         source_node=node_label,
         display_names=return_display_names,
         schema_ordered=return_ordered_by_schema,
     )
+    return [Node(node) for node in nodes]
```

Comment:
For all the endpoints, I have the same comment as I did for Node Properties...its confusing that its recorded as a list of nodes, bc nodes does not have an implicit type.. here it is `str` but in another context it could be a graph object... I feel like this nomenclature is obscuring what users should expect as an output. Im not familiar with this mode of typing so im not suggesting you change everything, but could you explain the choice further so I can understand it?

Reply:
I'm open to changing the name of the node class here. Its described here.. So just to clairify it is not a string, it is a class that has an attribute "name" which is a string. It's designed this way so that it can be extended in the future if we want to supply more informaiton about the node, and to be used by other endpoints.